diff --git a/python/beeai_framework/adapters/litellm/chat.py b/python/beeai_framework/adapters/litellm/chat.py
index b35acde3..8d01b0c9 100644
--- a/python/beeai_framework/adapters/litellm/chat.py
+++ b/python/beeai_framework/adapters/litellm/chat.py
@@ -231,6 +231,13 @@ class LiteLLMChatModel(ChatModel, ABC):
         )
 
     def _transform_output(self, chunk: ModelResponse | ModelResponseStream) -> ChatModelOutput:
+        if not chunk.choices:
+            usage = chunk.get("usage")  # type: ignore
+            return ChatModelOutput(
+                messages=[],
+                finish_reason="stop",
+                usage=ChatModelUsage(**usage.model_dump()) if usage else None,
+            )
         choice = chunk.choices[0]
         finish_reason = choice.finish_reason
         usage = chunk.get("usage")  # type: ignore
